client/dashboard/task-list/tasks/payments/index.js

```diff
@@ -6,30 +6,96 @@
 import { __ } from '@wordpress/i18n';
 import { Fragment, Component } from '@wordpress/element';
 import { compose } from '@wordpress/compose';
-import { filter, noop, keys, pickBy } from 'lodash';
+import { filter, noop, keys, pickBy, difference } from 'lodash';
 import { FormToggle, CheckboxControl } from '@wordpress/components';
 import { Button, TextControl } from 'newspack-components';
+import { withDispatch } from '@wordpress/data';
 
 /**
  * WooCommerce dependencies
  */
 import { getCountryCode } from 'dashboard/utils';
 import { Form, Card, Stepper, List } from '@woocommerce/components';
-import { getHistory, getNewPath } from '@woocommerce/navigation';
+import { getAdminLink, getHistory, getNewPath, getQuery } from '@woocommerce/navigation';
 
 /**
  * Internal dependencies
  */
 import withSelect from 'wc-api/with-select';
 import Plugins from '../steps/plugins';
+import Stripe from './stripe';
+import PayPal from './paypal';
 
 class Payments extends Component {
 	constructor() {
 		super( ...arguments );
+
+		let step = 'choose';
+		let showIndividualConfigs = false;
+		let forceStripeManual = false;
+
+		const query = getQuery();
+		const { options } = this.props;
+		const { woocommerce_onboarding_payments, woocommerce_stripe_settings } = options;
+
+		const installed = woocommerce_onboarding_payments && woocommerce_onboarding_payments.installed;
+		const methods =
+			( woocommerce_onboarding_payments && woocommerce_onboarding_payments.methods ) || [];
+		const configured =
+			( woocommerce_onboarding_payments && woocommerce_onboarding_payments.configured ) || [];
+
+		// Handle redirect back from Stripe
+		if (
+			query[ 'stripe-connect' ] &&
```

Comment:
@joshuatf I ended up refactoring the `constructor` quite a bit in https://github.com/woocommerce/woocommerce-admin/pull/2897 since the first pass here was a bit messy. It may be worth looking at the logic there instead. Hopefully that doesn't make the review too difficult. Let me know if it would be better to pull out just those changes back here.